### PR TITLE
#92240  Spring 3 upgrade

### PR DESCRIPTION
**Pull request by theo-ritense**

### CODE REVIEW

File: core/src/test/java/com/ritense/valtimo/web/rest/ChoiceFieldSecurityResourceTest.java [hunk 1]
```diff
@@ -58,7 +59,7 @@ void updateChoiceFieldAsUser() throws Exception {
     @Test
     @WithMockUser(username = USER_EMAIL, authorities = {ADMIN})
     void deleteChoiceFieldAsAdmin() throws Exception {
-        assertHttpStatus(DELETE, "/api/v1/choice-fields/1", INTERNAL_SERVER_ERROR);
+        assertHttpStatus(DELETE, "/api/v1/choice-fields/1", OK);
```
Comment (Klaas-Ritense) [inline core/src/test/java/com/ritense/valtimo/web/rest/ChoiceFieldSecurityResourceTest.java:62]: Strange

File: core/src/test/java/com/ritense/valtimo/web/rest/ChoiceFieldValueSecurityResourceTest.java [hunk 2]
```diff
@@ -58,7 +59,7 @@ void updateChoiceFieldValueAsUser() throws Exception {
     @Test
     @WithMockUser(username = USER_EMAIL, authorities = {ADMIN})
     void deleteChoiceFieldValueAsAdmin() throws Exception {
-        assertHttpStatus(DELETE, "/api/v1/choice-field-values/1", INTERNAL_SERVER_ERROR);
+        assertHttpStatus(DELETE, "/api/v1/choice-field-values/1", OK);
```
Comment (Klaas-Ritense) [inline core/src/test/java/com/ritense/valtimo/web/rest/ChoiceFieldValueSecurityResourceTest.java:62]: Strange

File: zgw/zaken-api/src/test/kotlin/com/ritense/zakenapi/mock/ZaakMockUrlProvider.kt [hunk 3]
```diff
@@ -24,6 +24,7 @@ import java.util.UUID
 
 @Service
 class ZaakMockUrlProvider : ZaakUrlProvider, ZaaktypeUrlProvider {
+    @Deprecated("Marked for removal since 10.5.0")
```
Comment (Klaas-Ritense) [inline zgw/zaken-api/src/test/kotlin/com/ritense/zakenapi/mock/ZaakMockUrlProvider.kt:27]: An `@Deprecated` inside a test class is strange

File: besluit/src/test/resources/config/application.yml
Comment (Klaas-Ritense) [inline besluit/src/test/resources/config/application.yml:1]: Implementations will have to make these changes as well, right?

File: authorization/src/main/kotlin/com/ritense/authorization/web/PermissionManagementResource.kt [hunk 4]
```diff
@@ -21,10 +21,12 @@ import com.ritense.authorization.deployment.PermissionDto
 import com.ritense.authorization.web.request.SearchPermissionsRequest
 import com.ritense.valtimo.contract.domain.ValtimoMediaType
 import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Controller
 import org.springframework.web.bind.annotation.PostMapping
 import org.springframework.web.bind.annotation.RequestBody
 import org.springframework.web.bind.annotation.RequestMapping
 
+@Controller
```
Comment (Klaas-Ritense) [inline authorization/src/main/kotlin/com/ritense/authorization/web/PermissionManagementResource.kt:30]: The `@Controller` in the Spring API Docs: "This annotation serves as a specialization of `@Component`, allowing for implementation classes to be autodetected through classpath scanning."

In other words: Valtimo will now try to create two beans: one because of the `@Controller` annotation, and another because this resource is also defined in the autoconfiguration.

File: objects-api/build.gradle [hunk 5]
```diff
@@ -38,7 +38,7 @@ dependencies {
     api "org.springframework.security:spring-security-core"
     api "org.springframework.boot:spring-boot"
     api "org.springframework.data:spring-data-commons"
-    api "javax.inject:javax.inject:${javaxInjectVersion}"
+    api "jakarta.inject:jakarta.inject-api:${javaxInjectVersion}"
```
Comment (Klaas-Ritense) [inline objects-api/build.gradle]: The name `javaxInjectVersion` is strange for a jakarta dependency

File: audit/src/main/java/com/ritense/audit/domain/AuditRecord.java [hunk 6]
```diff
@@ -53,14 +57,18 @@ public class AuditRecord implements Persistable<AuditRecordId> {
...
     @Column(name = "audit_event", columnDefinition = "json", updatable = false)
     @JsonView(AuditView.Public.class)
     private AuditEvent auditEvent;
 
     @Column(name = "document_id", updatable = false)
     private UUID documentId;
 
+    @Generated
+    @Column(name = "classname", updatable = false)
+    private String className;
```
Comment (Klaas-Ritense) [inline audit/src/main/java/com/ritense/audit/domain/AuditRecord.java:70]: Why is this needed? I don't see it being used anywhere

File: openzaak/src/main/kotlin/com/ritense/openzaak/plugin/OpenZaakUrlProvider.kt [hunk 7]
```diff
@@ -28,6 +28,7 @@ class OpenZaakUrlProvider(
     val zaakInstanceLinkService: ZaakInstanceLinkService,
     val zaakTypeLinkService: ZaakTypeLinkService
 ): ZaakUrlProvider, ZaaktypeUrlProvider {
+    @Deprecated("Marked for removal since 10.5.0")
```
Comment (Klaas-Ritense) [inline openzaak/src/main/kotlin/com/ritense/openzaak/plugin/OpenZaakUrlProvider.kt:31]: Why was this added?

File: build.gradle [hunk 8]
```diff
@@ -94,9 +94,11 @@ allprojects {
                 mavenBom "org.junit:junit-bom:${junitVersion}"
             }
             dependencies {
+                dependency "org.hibernate.orm:hibernate-core:${hibernateCoreVersion}"
                 dependency "org.yaml:snakeyaml:${snakeYamlVersionOverride}"
                 dependency "io.netty:netty-codec:${nettyCodecVersionOverride}"
                 dependency "org.liquibase:liquibase-core:${liquibaseVersion}"
+                dependency "com.rabbitmq:amqp-client:${amqpCLientVersion}"
```
Comment (Klaas-Ritense) [inline build.gradle:101]: Is adding rabbitmq to all projects necessary?

File: core/build.gradle [hunk 9]
```diff
@@ -88,23 +89,23 @@ dependencies {
...
     implementation "com.fasterxml.jackson.datatype:jackson-datatype-jsr310"
-    implementation "com.fasterxml.jackson.datatype:jackson-datatype-hibernate5"
+    implementation "com.fasterxml.jackson.datatype:jackson-datatype-hibernate6"
     implementation "com.fasterxml.jackson.dataformat:jackson-dataformat-xml"
     implementation "com.fasterxml.jackson.module:jackson-module-blackbird"
     implementation "com.fasterxml.jackson.module:jackson-module-parameter-names"
 
     // MyBatis (for camunda and our custom queries)
-    implementation "org.mybatis:mybatis:${mybatisVersion}"
+//    implementation "org.mybatis:mybatis:${mybatisVersion}"
```
Comment (Klaas-Ritense) [inline core/build.gradle]: Can this comment be removed?

File: core/src/test/java/com/ritense/valtimo/web/rest/ChoiceFieldSecurityResourceTest.java (hunk 1 above)
Comment (theo-ritense) [reply core/src/test/java/com/ritense/valtimo/web/rest/ChoiceFieldSecurityResourceTest.java:62]: This has been described in my comment on the story:
"SimpleJpaRepository.deleteById never followed spec to ignore not-found entities. This has been fixed in Hibernate 6". Also, the DELETE is now truly idempotent.

File: core/src/test/java/com/ritense/valtimo/web/rest/ChoiceFieldValueSecurityResourceTest.java (hunk 2 above)
Comment (theo-ritense) [reply core/src/test/java/com/ritense/valtimo/web/rest/ChoiceFieldValueSecurityResourceTest.java:62]: This has been described in my comment on the story:
"SimpleJpaRepository.deleteById never followed spec to ignore not-found entities. This has been fixed in Hibernate 6". Also, the DELETE is now truly idempotent.

File: zgw/zaken-api/src/test/kotlin/com/ritense/zakenapi/mock/ZaakMockUrlProvider.kt (hunk 3 above)
Comment (theo-ritense) [reply zgw/zaken-api/src/test/kotlin/com/ritense/zakenapi/mock/ZaakMockUrlProvider.kt:27]: Kotlin shows a warning when an implementation of a deprecated method or class is not also declaring the deprecation.

File: besluit/src/test/resources/config/application.yml
Comment (theo-ritense) [reply besluit/src/test/resources/config/application.yml:1]: Yes, this will be documented. I have added the keys to my 'to document'-comment in the story.

File: authorization/src/main/kotlin/com/ritense/authorization/web/PermissionManagementResource.kt (hunk 4 above)
Comment (theo-ritense) [reply authorization/src/main/kotlin/com/ritense/authorization/web/PermissionManagementResource.kt:30]: You are right, but the annotation is required since Spring 3, or it will not be used in the requestmapping. 
See my comment on the story: `@Controller or @RestController are now mandatory`

I don't know how we should fix this. Ideally, we should disable the component scanning I think.

File: audit/src/main/java/com/ritense/audit/domain/AuditRecord.java (hunk 6 above)
Comment (theo-ritense) [reply audit/src/main/java/com/ritense/audit/domain/AuditRecord.java:70]: The `classname` column was already there on the table. The content is generated by the DBMS. But the queries failed on Hibernate 6 because the field was not part of the entity class.

File: openzaak/src/main/kotlin/com/ritense/openzaak/plugin/OpenZaakUrlProvider.kt (hunk 7 above)
Comment (theo-ritense) [reply openzaak/src/main/kotlin/com/ritense/openzaak/plugin/OpenZaakUrlProvider.kt:31]: Kotlin shows a warning when an implementation of a deprecated method or class is not also declaring the deprecation.

File: build.gradle (hunk 8 above)
Comment (theo-ritense) [reply build.gradle:101]: That's not what this does. `dependencyManagement` specifies versions to use, IF they are declared in the project/module. I also declared the version explicitly on each module, but for some reason `app/gzac` chooses to ignore the declared version of our modules. Hence this 'hack'.

File: objects-api/build.gradle (hunk 5 above)
Comment (theo-ritense) [reply objects-api/build.gradle]: Renamed.

File: core/build.gradle (hunk 9 above)
Comment (theo-ritense) [reply core/build.gradle]: done.

File: audit/src/main/java/com/ritense/audit/repository/AuditRecordRepository.java [hunk 10]
```diff
@@ -40,7 +40,7 @@ List<AuditRecord> findAuditRecordsByEventAndProperty(
     );
 
     Page<AuditRecord> findByEventAndDocumentId(
-        List<Class<? extends AuditEvent>> eventTypes,
+        List<String> eventTypes,
```
Comment (marijnritense) [inline audit/src/main/java/com/ritense/audit/repository/AuditRecordRepository.java:43]: Please document this breaking change

File: authorization/src/main/kotlin/com/ritense/authorization/web/PermissionResource.kt [hunk 11]
```diff
@@ -32,6 +33,7 @@ import org.springframework.web.bind.annotation.RequestMapping
 import org.springframework.web.bind.annotation.RestController
 
 @RestController
+//@SkipComponentScan
```
Comment (ivo-ritense) [inline authorization/src/main/kotlin/com/ritense/authorization/web/PermissionResource.kt]: Why is this one commented out? Does this rely on component scanning?

File: audit/src/main/java/com/ritense/audit/repository/AuditRecordRepository.java (hunk 10 above)
Comment (theo-ritense) [reply audit/src/main/java/com/ritense/audit/repository/AuditRecordRepository.java:43]: noted in the story

File: authorization/src/main/kotlin/com/ritense/authorization/web/PermissionResource.kt (hunk 11 above)
Comment (theo-ritense) [reply authorization/src/main/kotlin/com/ritense/authorization/web/PermissionResource.kt]: That was unintended. Fixed.

File: contract/src/main/kotlin/com/ritense/valtimo/contract/annotation/SkipComponentScan.kt
Comment (Klaas-Ritense) [inline contract/src/main/kotlin/com/ritense/valtimo/contract/annotation/SkipComponentScan.kt:1]: Missing copyright header.

Nice solution though

File: contract/src/test/kotlin/com/ritense/valtimo/contract/annotation/SkipComponentScanIntTest.kt
Comment (Klaas-Ritense) [inline contract/src/test/kotlin/com/ritense/valtimo/contract/annotation/SkipComponentScanIntTest.kt:1]: Missing copyright header

File: contract/src/main/kotlin/com/ritense/valtimo/contract/annotation/SkipComponentScan.kt
Comment (theo-ritense) [reply contract/src/main/kotlin/com/ritense/valtimo/contract/annotation/SkipComponentScan.kt:1]: fixed

File: contract/src/test/kotlin/com/ritense/valtimo/contract/annotation/SkipComponentScanIntTest.kt
Comment (theo-ritense) [reply contract/src/test/kotlin/com/ritense/valtimo/contract/annotation/SkipComponentScanIntTest.kt:1]: fixed

**Review by ivo-ritense — DISMISSED** [review]

**Review by Klaas-Ritense — DISMISSED** [review]

**Review by ivo-ritense — APPROVED** [review]
